### PR TITLE
Change Citadel CI: from Bionic to Focal and change name from ign_ to use gz_

### PR DESCRIPTION
**Pull request by j-rivero**
Two main changes in our Citadel CI jobs:

* Instead of using Bionic which is EOL for keeping the CI running on Citadel, use Focal which is still supported. 
* Since we are going to change CI job names and I'm going to use the opportunity to also include the rename of `ign_` ot `gz_`. Keeping `ign_` is confusing the developer team with old  jobs and is going to be hard to configure in repositories for accepting CI jobs in each branch. This way the naming schema should use all `gz_` for all the collections.

This PR also fixes existing problems with Windows CI on Citadel that it is relying on having a `gz-` prefix name to overwrite the colcon workspace checkout.

### CODE REVIEW

**Comment by j-rivero** [comment]
jobs generated:
```diff
diff -ur /tmp/current_log_generated/jobs.txt /tmp/pr_log_generated/jobs.txt
--- /tmp/current_log_generated/jobs.txt	2023-11-28 18:28:57.829940876 +0000
+++ /tmp/pr_log_generated/jobs.txt	2023-11-28 18:27:34.254754364 +0000
@@ -1,49 +1,49 @@
 branch_ci __upcoming__ gz_tools-ci-main-homebrew-amd64
 branch_ci __upcoming__ gz_tools-ci-main-jammy-amd64
 branch_ci __upcoming__ gz_tools-main-win
-branch_ci citadel ign_cmake-2-win
-branch_ci citadel ign_cmake-ci-ign-cmake2-bionic-amd64
-branch_ci citadel ign_cmake-ci-ign-cmake2-homebrew-amd64
-branch_ci citadel ign_common-3-win
-branch_ci citadel ign_common-ci-ign-common3-bionic-amd64
-branch_ci citadel ign_common-ci-ign-common3-homebrew-amd64
-branch_ci citadel ign_fuel_tools-4-win
-branch_ci citadel ign_fuel_tools-ci-ign-fuel-tools4-bionic-amd64
-branch_ci citadel ign_fuel_tools-ci-ign-fuel-tools4-homebrew-amd64
-branch_ci citadel ign_gazebo-3-win
-branch_ci citadel ign_gazebo-ci-ign-gazebo3-bionic-amd64
-branch_ci citadel ign_gazebo-ci-ign-gazebo3-homebrew-amd64
-branch_ci citadel ign_gui-3-win
-branch_ci citadel ign_gui-ci-ign-gui3-bionic-amd64
-branch_ci citadel ign_gui-ci-ign-gui3-homebrew-amd64
-branch_ci citadel ign_launch-2-win
-branch_ci citadel ign_launch-ci-ign-launch2-bionic-amd64
-branch_ci citadel ign_launch-ci-ign-launch2-homebrew-amd64
-branch_ci citadel ign_math-6-win
-branch_ci citadel ign_math-ci-ign-math6-bionic-amd64
-branch_ci citadel ign_math-ci-ign-math6-homebrew-amd64
-branch_ci citadel ign_msgs-5-win
-branch_ci citadel ign_msgs-ci-ign-msgs5-bionic-amd64
-branch_ci citadel ign_msgs-ci-ign-msgs5-homebrew-amd64
-branch_ci citadel ign_physics-2-win
-branch_ci citadel ign_physics-ci-ign-physics2-bionic-amd64
-branch_ci citadel ign_physics-ci-ign-physics2-homebrew-amd64
-branch_ci citadel ign_plugin-1-win
-branch_ci citadel ign_plugin-ci-ign-plugin1-bionic-amd64
-branch_ci citadel ign_plugin-ci-ign-plugin1-homebrew-amd64
-branch_ci citadel ign_rendering-3-win
-branch_ci citadel ign_rendering-ci-ign-rendering3-bionic-amd64
-branch_ci citadel ign_rendering-ci-ign-rendering3-homebrew-amd64
-branch_ci citadel ign_sensors-3-win
-branch_ci citadel ign_sensors-ci-ign-sensors3-bionic-amd64
-branch_ci citadel ign_sensors-ci-ign-sensors3-homebrew-amd64
-branch_ci citadel ign_tools-1-win
-branch_ci citadel ign_tools-ci-ign-tools1-bionic-amd64
-branch_ci citadel ign_tools-ci-ign-tools1-homebrew-amd64
-branch_ci citadel ign_transport-8-win
-branch_ci citadel ign_transport-ci-ign-transport8-bionic-amd64
-branch_ci citadel ign_transport-ci-ign-transport8-homebrew-amd64
-branch_ci citadel sdformat-ci-sdf9-bionic-amd64
+branch_ci citadel gz_cmake-ci-ign-cmake2-focal-amd64
+branch_ci citadel gz_cmake-ci-ign-cmake2-homebrew-amd64
+branch_ci citadel gz_cmake-ign-cmake2-win
+branch_ci citadel gz_common-ci-ign-common3-focal-amd64
+branch_ci citadel gz_common-ci-ign-common3-homebrew-amd64
+branch_ci citadel gz_common-ign-common3-win
+branch_ci citadel gz_fuel_tools-ci-ign-fuel-tools4-focal-amd64
+branch_ci citadel gz_fuel_tools-ci-ign-fuel-tools4-homebrew-amd64
+branch_ci citadel gz_fuel_tools-ign-fuel-tools4-win
+branch_ci citadel gz_gui-ci-ign-gui3-focal-amd64
+branch_ci citadel gz_gui-ci-ign-gui3-homebrew-amd64
+branch_ci citadel gz_gui-ign-gui3-win
+branch_ci citadel gz_launch-ci-ign-launch2-focal-amd64
+branch_ci citadel gz_launch-ci-ign-launch2-homebrew-amd64
+branch_ci citadel gz_launch-ign-launch2-win
+branch_ci citadel gz_math-ci-ign-math6-focal-amd64
+branch_ci citadel gz_math-ci-ign-math6-homebrew-amd64
+branch_ci citadel gz_math-ign-math6-win
+branch_ci citadel gz_msgs-ci-ign-msgs5-focal-amd64
+branch_ci citadel gz_msgs-ci-ign-msgs5-homebrew-amd64
+branch_ci citadel gz_msgs-ign-msgs5-win
+branch_ci citadel gz_physics-ci-ign-physics2-focal-amd64
+branch_ci citadel gz_physics-ci-ign-physics2-homebrew-amd64
+branch_ci citadel gz_physics-ign-physics2-win
+branch_ci citadel gz_plugin-ci-ign-plugin1-focal-amd64
+branch_ci citadel gz_plugin-ci-ign-plugin1-homebrew-amd64
+branch_ci citadel gz_plugin-ign-plugin1-win
+branch_ci citadel gz_rendering-ci-ign-rendering3-focal-amd64
+branch_ci citadel gz_rendering-ci-ign-rendering3-homebrew-amd64
+branch_ci citadel gz_rendering-ign-rendering3-win
+branch_ci citadel gz_sensors-ci-ign-sensors3-focal-amd64
+branch_ci citadel gz_sensors-ci-ign-sensors3-homebrew-amd64
+branch_ci citadel gz_sensors-ign-sensors3-win
+branch_ci citadel gz_sim-ci-ign-gazebo3-focal-amd64
+branch_ci citadel gz_sim-ci-ign-gazebo3-homebrew-amd64
+branch_ci citadel gz_sim-ign-gazebo3-win
+branch_ci citadel gz_tools-ci-ign-tools1-focal-amd64
+branch_ci citadel gz_tools-ci-ign-tools1-homebrew-amd64
+branch_ci citadel gz_tools-ign-tools1-win
+branch_ci citadel gz_transport-ci-ign-transport8-focal-amd64
+branch_ci citadel gz_transport-ci-ign-transport8-homebrew-amd64
+branch_ci citadel gz_transport-ign-transport8-win
+branch_ci citadel sdformat-ci-sdf9-focal-amd64
 branch_ci citadel sdformat-ci-sdf9-homebrew-amd64
 branch_ci citadel sdformat-sdf9-win
 branch_ci fortress gz_cmake-ci-ign-cmake2-focal-amd64
```

**Review by Crola1702 — APPROVED** [review]
Looks good. Just one minor comment.

Also, bumping to `focal` and changing to `gz_` will collide with some fortress packages.

E.g.: `citadel gz_cmake-ci-ign-cmake2-focal-amd64` and `fortress gz_cmake-ci-ign-cmake2-focal-amd64` (in fact, this job already exists: [link](https://build.osrfoundation.org/job/gz_cmake-ci-ign-cmake2-focal-amd64/)).

I'm ok with that, as specifically, for these jobs that's the way the Gazebo dashboard should show the jobs.

**Comment by j-rivero** [comment]
> Also, bumping to `focal` and changing to `gz_` will collide with some fortress packages.
> 
> E.g.: `citadel gz_cmake-ci-ign-cmake2-focal-amd64` and `fortress gz_cmake-ci-ign-cmake2-focal-amd64` (in fact, this job already exists: [link](https://build.osrfoundation.org/job/gz_cmake-ci-ign-cmake2-focal-amd64/)).
> 
> I'm ok with that, as specifically, for these jobs that's the way the Gazebo dashboard should show the jobs.

This is expected since the same library and version is included in both collections.

File: jenkins-scripts/dsl/gz-collections.yaml [hunk 1]
```diff
@@ -583,20 +560,6 @@ ci_configs:
...
-      - ign-msgs
-      - ign-physics
-      - ign-rendering
-      - ign-sensors
-      - ign-tools
-      - ign-transport
-      - ign-utils
       - sdformat
 packaging_configs:
   - name: bionic
```
[the remark this answers is not in the document]
Comment (j-rivero) [reply jenkins-scripts/dsl/gz-collections.yaml]: Removed in [916ffd9](https://github.com/gazebo-tooling/release-tools/pull/1083/commits/916ffd909af2acc22c9aca04ddcd3ab9af522ea3). I think it is currently not used.

**Pull request merged**